src/Uno.UI/UI/Xaml/Controls/Grid/Grid.cs

```diff
@@ -186,18 +190,26 @@ private void OnRowDefinitions_CollectionChanged(object sender, IVectorChangedEve
 
 		private void DisposeAllRowsColumnsSubscriptions()
 		{
-			DisposeDefinitionsSubscriptions(_rowsSubscriptions);
-			DisposeDefinitionsSubscriptions(_columnsSubscriptions);
+			DisposeDefinitionsSubscriptions(_rowsHeightSubscriptions);
+			DisposeDefinitionsSubscriptions(_rowsMinHeightSubscriptions);
+			DisposeDefinitionsSubscriptions(_rowsMaxHeightSubscriptions);
+			DisposeDefinitionsSubscriptions(_columnsWidthSubscriptions);
+			DisposeDefinitionsSubscriptions(_columnsMinWidthSubscriptions);
+			DisposeDefinitionsSubscriptions(_columnsMaxWidthSubscriptions);
 		}
 
 		private void ObserveRowDefinitions(List<RowDefinition> definitions)
 		{
-			ObserveDefinitions(_rowsSubscriptions, definitions, RowDefinition.HeightProperty, OnGridDefinitionChanged);
+			ObserveDefinitions(_rowsHeightSubscriptions, definitions, RowDefinition.HeightProperty, OnGridDefinitionChanged);
```

Comment:
There should be simple tests for those as well.

Reply:
right!